src/vg-tooltip.js

```diff
@@ -0,0 +1,399 @@
+        tooltipUtil.init(event, item, options);
+      });
+
+      // update tooltip position on mouse move
+      // (important for large marks e.g. bars)
+      vgView.on("mousemove", function(event, item) {
+        tooltipUtil.update(event, item, options);
+      });
+
+      // clear tooltip
+      vgView.on("mouseout", tooltipUtil.clear);
+    }
+}());
+
+/**
+ * Export Vega-Lite Tooltip API: vlTooltip.linkToView(vgView, vlSpec, options)
+ * options can specify data fields to show in the tooltip and can
+ * overwrite data formats in vlSpec
+ */
+var vlTooltip = (function() {
+
+  // supplement options with timeUnit and numberFormat from vlSpec
+  function supplementOptions (options, vlSpec) {
+    // supplement numberFormat
+    if (vlSpec.config && vlSpec.config.numberFormat) {
+      options.numberFormat = vlSpec.config.numberFormat;
+    }
+
+    // supplement timeUnit
+    vl.spec.fieldDefs(vlSpec).forEach(function(channel) {
+      if (channel.timeUnit) {
+        if (!options.fieldsWithTimeUnit) {
+          options.fieldsWithTimeUnit = [];
+        }
+        try {
+          // TODO(zening): rename field because VL renames field
+          // e.g. date --> month_date, date --> year_date
+          var renamedField = channel.timeUnit + '_' + channel.field;
+          options.fieldsWithTimeUnit.push({field: renamedField, timeUnit: channel.timeUnit});
+        }
+        catch (error) {
+          console.error('[VgTooltip] Error parsing Vega-Lite timeUnit: ' + error);
+        }
+      }
+    });
+
+    return options;
+  }
+
+  return function(vgView, vlSpec, options) {
+      if (!options) {
+        options = {};
+      }
+
+      options = supplementOptions(options, vlSpec);
+
+      // initialize tooltip with data
+      vgView.on("mouseover", function(event, item) {
+        tooltipUtil.init(event, item, options);
+      });
+
+      // update tooltip position on mouse move
+      // (important for large marks e.g. bars)
+      vgView.on("mousemove", function(event, item) {
+        tooltipUtil.update(event, item, options);
+      });
+
+      // clear tooltip
+      vgView.on("mouseout", tooltipUtil.clear);
+    }  
+}());
+
+/**
+ * Export common utilities: init, update and clear
+ * init(): initialize tooltip with data
+ * update(): update tooltip as mouse moves
+ * clear(): clear tooltip data
+ */
+var tooltipUtil = (function() {
+
+  /* Decide if a chart element deserves tooltip */
+  function shouldShowTooltip (item) {
+    if (!item || !item.datum) return false;
+
+    // avoid showing tooltip for facet's background
+    if (item.datum._facetID) return false;
+
+    // avoid showing tooltip for axis title and labels
+    if (!item.datum._id) return false;
+
+    return true;
+  }
+
+  /**
+   * Update tooltip position
+   * Default position is 10px right of and 10px below the cursor. This can be
+   * overwritten by options.
+   */
+  function updatePosition (event, options) {
```

Comment:
Can you remove all spaces between function name and `(`
e.g., `updatePosition(...)` ?